thumbor/engines/pil.py

```diff
@@ -342,6 +348,25 @@ def convert_to_grayscale(self, update_image=True, with_alpha=True):
             self.image = image
         return image
 
+    def has_transparency(self):
+        has_transparency = 'A' in self.image.mode or 'transparency' in self.image.info
+        if has_transparency and numpy:
+            # If the image has alpha channel and numpy is installed
+            # we check for any pixels that are not opaque (255)
+            img = self.image.convert('RGBA')
```

Comment:
I think this could be done with pure Pillow.
```python
sum(img.convert('RGBA').getband('A') # getband is pillow 4.3+
              .point(lambda x: 0 if x else 1)  # return 1 if pixel is transparent, 0 otherwise
              .getdata())
```
something like that (untested)

Reply:
```python
img.convert('RGBA').getchannel('A').getextrema()
```

that returns tuple with min and max pixel values [and iterates in C](https://github.com/python-pillow/Pillow/blob/81c88b1ead5343d9778a82977c9c6c1150fd9738/src/libImaging/GetBBox.c#L109)

Reply:
So, you think pushing bytes from pillow C-level to numpy C-level would be faster than just calling C-func on pillow?

Reply:
Nope, I asked @caeugusmao to change to the way you described (`img.convert('RGBA').getchannel('A').getextrema()`) since it will be a lot faster. What I said is that, in your first solution, `sum` would have to iterate over all pixels, wouldn't it? And that probably is slower than iterating over in C, at least for large images.

Once again, the last solution you gave is way better. I wasn't aware of `getextrema`.